script/hassfest/icons.py

```diff
@@ -0,0 +1,106 @@
+"""Validate integration icon translation files."""
+from __future__ import annotations
+
+from typing import Any
+
+import orjson
+import voluptuous as vol
+from voluptuous.humanize import humanize_error
+
+import homeassistant.helpers.config_validation as cv
+
+from .model import Config, Integration
+from .translations import translation_key_validator
+
+
+def icon_value_validator(value: Any) -> str:
+    """Validate that the icon is a valid icon."""
+    value = cv.string_with_no_html(value)
+    if not value.startswith("mdi:"):
+        raise vol.Invalid(
+            "The icon needs to be a valid icon from Material Design Icons and start with `mdi:`"
+        )
+    return str(value)
+
+
+def icon_schema(integration_type: str) -> vol.Schema:
+    """Create a icon schema."""
+    if integration_type == "entity":
+        return vol.Schema(
+            {
+                vol.Optional("entity_component"): cv.schema_with_slug_keys(
```

Comment:
To reduce code duplication we could use `partial` for `cv.schema_with_slug_keys(` with `slug_validator=translation_key_validator,` as it is used multiple times here